iit/model_pairs/iit_behavior_model_pair.py

```diff
@@ -108,32 +109,38 @@ def run_eval_step(
 
         # compute IIT loss and accuracy
         label_idx = self.get_label_idxs()
-        hl_node = self.sample_hl_name()
-        hl_output, ll_output = self.do_intervention(base_input, ablation_input, hl_node)
-        hl_output.to(ll_output.device)
-        hl_output = hl_output[label_idx.as_index]
-        ll_output = ll_output[label_idx.as_index]
-        if self.hl_model.is_categorical():
-            loss = loss_fn(ll_output, hl_output)
-            if ll_output.shape == hl_output.shape:
-                # To handle the case when labels are one-hot
-                hl_output = t.argmax(hl_output, dim=-1)
-            top1 = t.argmax(ll_output, dim=-1)
-            accuracy = (top1 == hl_output).float().mean()
-            IIA = accuracy.item()
-        else:
-            loss = loss_fn(ll_output, hl_output)
-            IIA = ((ll_output - hl_output).abs() < atol).float().mean().item()
+        #don't just sample one HL node, compute IIA for all HL nodes and average.
+        iias = []
```

Comment:
This would increase the validation time for large tasks(eg: IOI) like crazy. Maybe would be better to keep it under a flag and optionally add another tqdm to the validation round.

Reply:
added "val_IIA_sampling" to training metrics with default value of "random" (can choose "all").